qiskit/qpy/__init__.py

```diff
@@ -126,6 +126,62 @@
 by ``num_circuits`` in the file header). There is no padding between the
 circuits in the data.
 
+.. _qpy_version_8:
+
+Version 8
+=========
+
+Version 8 adds support for handling a :class:`~.TranspileLayout` stored in the
+:attr:`.QuantumCircuit.layout` attribute. In version 8 immediately following the
+calibrations block at the end of the circuit payload there is now the
+``LAYOUT`` struct. This struct outlines the size of the three attributes of a
+:class:`~.TranspileLayout` class.
+
+LAYOUT
+------
+
+.. code-block:: c
+
+    struct {
+        int32_t initial_layout_size;
+        int32_t input_mapping_size;
+        int32_t final_layout_size;
+        uint32_t extra_registers;
+    }
```

Comment:
Ah, yeah I added that after I started writing the docs because as I was testing locally I needed to differentiate between no layout and an empty layout. I'll update this to reflect that.

As for the use of `_Bool` I guess yeah that's true it is 1 byte long, I took the type from: https://docs.python.org/3/library/struct.html#format-characters but I neglected to read the footnotes about `_Bool` a more accurate type is probably `char` based on that. I think we can go through and update the docs to better reflect that in a follow up PR.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/10148/commits/dd0efc149d73525ec53ca60681a89e617fb5fe84